resources/js/components/comments-controller.ts

```diff
@@ -0,0 +1,538 @@
+        this.state.votedCommentIds[storeMethod](comment.id);
+      })).fail(onError)
+      .always(action(() => {
+        delete(this.xhr[xhrKey]);
+      }));
+  }
+
+  readonly destroy = () => {
+    if (this.destroyed) return;
+
+    document.removeEventListener('turbolinks:before-cache', this.destroy);
+    abortXhrCollection(this.xhr);
+    this.stateStore();
+    this.destroyed = true;
+  };
+
+  getCommentableMeta(comment: Comment) {
+    return this.state.commentableMetaItems[`${comment.commentableType}-${comment.commentableId}`] ?? { title: '' };
+  }
+
+  getComments(ids: number[] | undefined) {
+    const ret = [];
+
+    for (const id of ids ?? []) {
+      const comment = this.state.comments[id];
+
+      if (comment != null) {
+        ret.push(comment);
+      }
+    }
+
+    return ret;
+  }
+
+  getUser(id: number | null | undefined) {
+    return id == null ? undefined : this.state.users[id];
+  }
+
+  isLoading(parent: Comment | null | undefined) {
+    return this.xhr[`load-${parent?.id ?? 0}`] != null;
+  }
+
+  isPosting(postParams: PostParams) {
+    return this.xhr[`${postParams.mode}-${postXhrKeyId(postParams)}`] != null;
+  }
+
+  isVoting(comment: Comment) {
+    return this.xhr[`vote-${comment.id}`] != null;
+  }
+
+  private addCommentId(comment: CommentJson, append: boolean) {
+    const parentId = comment.parent_id ?? 0;
+    this.state.commentIdsByParentId[parentId] ??= [];
+
+    // The `?? []` shouldn't ever happen.
+    const ids = this.state.commentIdsByParentId[parentId] ?? [];
+    const newId = comment.id;
+    if (!ids.includes(newId)) {
+      if (append) {
+        ids.push(newId);
+      } else {
+        ids.unshift(newId);
+      }
+    }
+  }
+
+  private isDeleting(comment: Comment) {
+    return this.xhr[`delete-${comment.id}`] != null;
+  }
+
+  @action
+  private loadBundle(bundle: CommentBundleJson, append = true, initial = false) {
+    if (initial) {
+      this.state.commentIdsByParentId[0] = bundle.comments.map((comment) => comment.id);
+      this.state.sort = bundle.sort;
+    } else {
+      bundle.comments.forEach((comment) => this.addCommentId(comment, append));
+    }
+
+    bundle.included_comments.forEach((comment) => this.addCommentId(comment, true));
+    if (bundle.pinned_comments != null) {
+      this.state.pinnedCommentIds = bundle.pinned_comments.map((comment) => comment.id);
+    }
+    bundle.user_votes.forEach((v) => this.state.votedCommentIds.add(v));
+
+    this.state.isFollowing = bundle.user_follow;
+    this.state.hasMore[bundle.has_more_id] = bundle.has_more;
+    if (bundle.top_level_count != null && bundle.total != null) {
+      this.state.topLevelCount = bundle.top_level_count;
+      this.state.total = bundle.total;
+    }
+
+    for (const comment of [...bundle.comments, ...bundle.included_comments, ...bundle.pinned_comments]) {
+      const id = comment.id;
+      if (this.state.comments[id]?.updatedAt !== comment.updated_at) {
```

Comment:
The bundle returned when replying uses the original timestamp instead of the updated one, so the reply count doesn't get updated correctly (also means first reply to a comment doesn't show).

Reply:
more like the updated_at column wasn't touched at all when incrementing...